pages/discover.js

```diff
@@ -6,31 +6,84 @@ import NextLink from "next/link";
 import Header from "../components/header";
 import Body from "../components/body";
 import { getJSON } from "../shims/ajax";
+import {createGlobalStyle} from "styled-components"
 
 const Authors = props => (
   <span className="authors">
     {props.authors.map(author => author.name).join(", ")}
   </span>
 );
 
+const TagStyle = createGlobalStyle`
+  .tag {
+    display: inline-block;
+    padding: 0.2em 0.9em;
+    margin: 0 0.5em 0.5em 0;
+    white-space: nowrap;
+    background-color: #f1f8ff;
+    border-radius: 3px;
+    color: #0366d6;
+    text-decoration: none;
+  }
+  tag:hover {
+    background-color: #ddeeff;
+  }
+`
+
```

Comment:
Instead of making these globals, I'd probably make elements like `Post`, `PostMetadata`, and `PostTags`, then use those compositionally.

```jsx
  <Post> {/* was <div className="post"> */}
    <PostThumb>
      <img
        alt={`preview for ${props.path}`}
        src={
          props.image ? props.image : "https://icon.now.sh/library_books/ccc/64"
        }
        height="64"
      />
    </PostThumb>
    <PostSummary>
      <PostTitle>
        <NextLink
          href={{ pathname: "/view", query: { viewPath: props.path } }}
          as={"/view/" + props.path}
        >
          <a>{props.metadata.title ? props.metadata.title : props.name}</a>
        </NextLink>
      </PostTitle>
      <PostMetadata>
        <span>
          Last modified <TimeAgo date={props.last_modified} />
        </span>
        {` `}
        {props.metadata.authors ? (
          <span>
            by <Authors authors={props.metadata.authors} />
          </span>
        ) : null}
      </PostMetadata>
      <PostDescription>
        <p>{props.metadata.nteract.description}</p>
      </PostDescription>
      <PostTags>
        {props.metadata.nteract.tags.map(tag => (
          <Tag key={tag}>{tag}</Tag>
        ))}
      </PostTags>
    </Post>
```

Reply:
I feel like that is a much bigger refactor than is needed to simply shift over the current functionality over to styled-components (even if it's not what we want to do in the long run).

I'd prefer to focus on the typescript side of this for this PR.